includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -976,61 +976,36 @@ public static function add_nav_menu_styles( $args = array() ) {
 	}
 
 	/**
-	 * Ensure that JS-only nav menu styles apply to AMP as well since even though scripts are not allowed, there are AMP-bind implementations.
+	 * Ensure that support is added for nav menu toggles that are typically JS-only.
 	 *
 	 * @since 1.0
+	 * @since 1.1 Made method static so it acts as a buffering hook for adjusting theme support instead of a post-sanitizer.
 	 *
 	 * @param array $args Args.
 	 */
-	public function add_nav_menu_toggle( $args = array() ) {
+	public static function add_nav_menu_toggle( $args = array() ) {
 		$args = array_merge(
 			self::get_theme_config( get_template() ),
 			$args
 		);
 
-		$nav_el    = $this->dom->getElementById( $args['nav_container_id'] );
-		$button_el = $this->xpath->query( $args['menu_button_xpath'] )->item( 0 );
-		if ( ! $nav_el ) {
-			if ( $button_el ) {
-
-				// Remove the button since it won't be used.
-				$button_el->parentNode->removeChild( $button_el );
-			}
+		if ( empty( $args ) ) {
 			return;
 		}
 
-		if ( ! $button_el ) {
-			return;
+		$support = AMP_Theme_Support::get_theme_support_args();
+		if ( ! is_array( $support ) ) {
+			$support = array();
 		}
 
-		$state_id = 'navMenuToggledOn';
-		$expanded = false;
-
-		$nav_el->setAttribute(
-			AMP_DOM_Utils::get_amp_bind_placeholder_prefix() . 'class',
-			sprintf(
-				"%s + ( $state_id ? %s : '' )",
-				wp_json_encode( $nav_el->getAttribute( 'class' ) ),
-				wp_json_encode( ' ' . $args['nav_container_toggle_class'] )
-			)
+		$support['nav_menu_toggle'] = array(
+			'nav_container_id'             => $args['nav_container_id'],
+			'nav_menu_button_xpath'        => $args['menu_button_xpath'],
+			'nav_container_toggle_class'   => $args['nav_container_toggle_class'],
+			'nav_menu_button_toggle_class' => $args['menu_button_toggle_class'],
```

Comment:
Why not just use the same keys? This could just be:

```php
$support['nav_menu_toggle'] = $args;
```

Reply:
Sounds good, I'll change it.